cli/src/commands/help.rs

```diff
@@ -47,3 +79,74 @@ pub(crate) fn cmd_help(
 
     Err(command_error::cli_error(help_err))
 }
+
+#[derive(Clone)]
+struct Keyword {
+    name: &'static str,
+    description: &'static str,
+    content: &'static str,
+}
+
+// TODO: Add all documentation to keywords
+//
+// Maybe adding some code to build.rs to find all the docs files and build the
+// `KEYWORDS` at compile time.
+//
+// It would be cool to follow the docs hierarchy somehow.
+//
+// One of the problems would be `config.md`, as it has the same name as a
+// subcommand.
+//
+// TODO: Find a way to render markdown using ANSI escape codes.
+//
+// Maybe we can steal some ideas from https://github.com/martinvonz/jj/pull/3130
+const KEYWORDS: &[Keyword] = &[
+    Keyword {
+        name: "revsets",
+        description: "A functional language for selecting a set of revision",
+        content: include_str!("../../../docs/revsets.md"),
+    },
+    Keyword {
+        name: "tutorial",
+        description: "Show a tutorial to get started with jj",
+        content: include_str!("../../../docs/tutorial.md"),
+    },
+];
+
+fn find_keyword(name: &str) -> Option<&Keyword> {
+    KEYWORDS.iter().find(|keyword| keyword.name == name)
+}
+
+fn write_keyword_list(mut formatter: Box<dyn Formatter + '_>) -> io::Result<()> {
+    let keyword_name_max_len = KEYWORDS
+        .iter()
+        .map(|keyword| keyword.name.len())
+        .max()
+        .unwrap();
+
+    formatter.push_label("bold_text")?;
+    formatter.push_label("underlined_text")?;
+    writeln!(formatter, "Help Keywords:")?;
+    formatter.pop_label()?;
+    formatter.pop_label()?;
+    for keyword in KEYWORDS {
+        formatter.push_label("bold_text")?;
+        write!(formatter, "  {:<keyword_name_max_len$}  ", keyword.name)?;
+        formatter.pop_label()?;
```

Comment:
`formatter.labeled("help_keyword")` or `formatter.with_label("help", |formatter| { ... formatter.labeled("keyword") .. })`?

Labels usually describe the content, not the style.

Reply:
BTW, another idea is to use `PossibleValue::new(k.name).help(k.description)` and suggest `jj help --help` instead of implementing `jj help --list-keywords`.
https://docs.rs/clap/latest/clap/builder/struct.PossibleValue.html